crates/bevy_ui/src/update.rs

```diff
@@ -37,35 +37,46 @@ fn update_clipping(
     entity: Entity,
     clip: Option<Rect>,
 ) {
-    let (node, global_transform, style, calculated_clip) = node_query.get_mut(entity).unwrap();
-    // Update this node's CalculatedClip component
-    match (clip, calculated_clip) {
+    let (node, global_transform, style, old_clip) = node_query.get_mut(entity).unwrap();
+
+    // Update current node's CalculatedClip component
+    match (old_clip, clip) {
         (None, None) => {}
-        (None, Some(_)) => {
+        (Some(_), None) => {
             commands.entity(entity).remove::<CalculatedClip>();
         }
-        (Some(clip), None) => {
+        (None, Some(clip)) => {
             commands.entity(entity).insert(CalculatedClip { clip });
         }
-        (Some(clip), Some(mut old_clip)) => {
+        (Some(mut old_clip), Some(clip)) => {
             if old_clip.clip != clip {
                 *old_clip = CalculatedClip { clip };
             }
         }
     }
 
-    // Calculate new clip for its children
+    // Calculate new clip rectangle for children nodes
     let children_clip = match style.overflow {
+        // When `Visible`, children will be visible even when they are outside
+        // the current node's "area". In this case they inherit the current
+        // node's parent clip. If an ancestor is set as `Hidden`, that clip will
+        // be used; otherwise this will be `None`.
         Overflow::Visible => clip,
         Overflow::Hidden => {
+            // Calculate current node clip rectangle from: posisition + calculated_size
             let node_center = global_transform.translation().truncate();
```

Comment:
There is a helper function for this now:
```suggestion
            let node_clip = node.logical_rect(global_transform);
```